### PR TITLE
automating test case 832450

### PR DESCRIPTION
**Pull request by kkrishnareddy**
Automating test case 832450

### CODE REVIEW

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+
+//Non Joined AcquireToken test with MSAL and Broker
```
Comment (jaiswalh) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:49]: Javadoc format.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+//https:identitydivision.visualstudio.com/DevEx/_workitems/edit/832450
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:52]: The link format is incorrect

Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:52]:
```suggestion
// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/832450
```

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                new OnInteractionRequired() {
+                    @Override
+                    public void handleUserInteraction() {
+                        final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                                .prompt(PromptParameter.SELECT_ACCOUNT)
+                                .loginHint(mLoginHint)
+                                .sessionExpected(false)
+                                .consentPageExpected(false)
+                                .speedBumpExpected(false)
+                                .broker(getBroker())
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:83]:
```suggestion
                                .broker(mBroker)
```

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        // forwarding time for 1 day.
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+
+        // acquiring token silently
+        final CountDownLatch refreshTokenLatch = new CountDownLatch(1);
+
+        final AcquireTokenSilentParameters refreshSilentParameters = new AcquireTokenSilentParameters.Builder()
+                .forAccount(account)
+                .fromAuthority(account.getAuthority())
+                .forceRefresh(true)
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:121]:
```suggestion
                .forceRefresh(false)
```

We don't want to force refresh here (the test didn't ask for it)

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                .withScopes(Arrays.asList(mScopes))
+                .withCallback(successfulSilentCallback(silentLatch))
+                .build();
+
+        mApplication.acquireTokenSilentAsync(silentParameters);
+        silentLatch.await();
+
+        // forwarding time for 1 day.
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:114]: We also want to do a Thread.sleep for 30 seconds here to expire the in-memory throttling cache

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    @Override
+    public int getConfigFileResourceId() {
+        return R.raw.msal_config_default;
+    }
+
+    @NonNull
+    @Override
+    public ITestBroker getBroker() {
+        return new BrokerMicrosoftAuthenticator();
+    }
```
Comment (shahzaibj) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:149]: You no longer need to override this

**Review by shahzaibj — COMMENTED** [review]
Left some comments and suggestions

**Comment by shahzaibj** [comment]
@kkrishnareddy Are all comments addressed on this one?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java (hunk 6 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java]: removed getBroker() function from testcase

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java (hunk 5 above)
Comment (kkrishnareddy) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase832450.java:116]: added 30 seconds of Thread.sleep.

**Pull request closed without merging**